File: doc-examples/example-protobuf-kotlin/src/main/kotlin/example/SensorReading.kt
package example

import io.micronaut.serde.annotation.Serdeable
import io.micronaut.serde.protobuf.annotation.ProtoField
import io.micronaut.serde.protobuf.annotation.ProtoType

// tag::clazz[]
@Serdeable
data class SensorReading(
    @field:ProtoField(1) val deviceId: String,
    @field:ProtoField(2) val recordedAt: Long,
    @field:ProtoField(value = 3, type = ProtoType.SINT32) val temperatureMilliCelsius: Int,   // <1>
    @field:ProtoField(4) val signalStrengths: List<Int>?,                                     // <2>
    @field:ProtoField(5) val location: Coordinates?,                                          // <3>
    @field:ProtoField(6) val batteryLow: Boolean
)
// end::clazz[]

File: doc-examples/example-protobuf-python/src/main/python/example/SensorReading.py
from dataclasses import dataclass
from typing import Annotated

from java.lang import Long

from micronaut.serde.annotation import Serdeable
from micronaut.serde.protobuf.annotation import ProtoField, ProtoType

from example.Coordinates import Coordinates

# tag::clazz[]
@Serdeable
@dataclass(frozen=True)
class SensorReading:
    deviceId: Annotated[str, ProtoField(1)]
    recordedAt: Annotated[Long, ProtoField(2)]
    temperatureMilliCelsius: Annotated[int, ProtoField(value=3, type=ProtoType.SINT32)]   # <1>
    signalStrengths: Annotated[list[int] | None, ProtoField(4)]                            # <2>
    location: Annotated[Coordinates | None, ProtoField(5)]                                 # <3>
    batteryLow: Annotated[bool, ProtoField(6)]
# end::clazz[]
